pkg/cli/commands_test.go

```
@@ -0,0 +1,75 @@
+	v.SetConfigType("yaml")
+
+	// Simulate setting values in Viper (normally read from a file)
+	v.Set("port", 8080)
+	v.Set("host", "localhost")
+	v.Set("with-dash", "foobar")
+
+	// Use the generic function to load config
+	config, err := makeGenericConfiguration[AppConfig](v)
+	require.NoError(t, err)
+
+	fmt.Printf("Loaded Config: %+v\n", config)
```

golangci-lint (1.62.2)

use of `fmt.Printf` forbidden by pattern `^(fmt\.Print(|f|ln)|print|println)$`

(forbidigo)